### DISCUSSION TITLE
HELP - no clues on tuple input error for combined channels

### DISCUSSION THREAD

**Discussion by julianajose** [nextflow-io/nextflow#3253] [Q&A]
Hi everyone,
I really need some help with the tuple input produced by two channels combined. I think I have followed the documentation instructions, and after all the errors I got I reduced the code to the minimum. It is looking like this:
`
#!/usr/bin/env nextflow

readcount_out   = params.work_dir + '02.CollectReadCounts'
cnv_out               = params.work_dir + '04.CNVcall'

intervals = Channel.fromPath(cnv_out + "/*-model")
sample = Channel.fromPath(readcount_out + "/*.tsv")
ChannelCombined=sample.combine(intervals)

process GatkCohortCNVcall {
    executor 'slurm'
    cpus params.gatk_threads

    input:
         tuple val(sample), path(intervals) from ChannelCombined
    
   script:
    """
    echo $sample $intervals
    """    
}
`
No matter if I use val(), path(), or file() in tuple inputs, I get the same error:

ERROR ~ No signature of method: groovyx.gpars.dataflow.DataflowQueue.getFileSystem() is applicable for argument types: () values: []

Anyone have clues about what’s going on?

**Comment by jemunro** [comment]
What nextflow version and DSL are you using? based on the process definition I assume you are using DSL1.

What is the expected full path to the files in the channels `intervals` and `sample`?

I suggest you use the optional `checkIfExists` argument for fromPath to make sure you've given the correct path to the files:
```nextflow
intervals = Channel.fromPath(cnv_out + "/-model", checkIfExists: true)
sample = Channel.fromPath(readcount_out + "/.tsv", checkIfExists: true)
```
see https://www.nextflow.io/docs/latest/channel.html#frompath

**Reply by julianajose** [reply]
Thank you so much for the suggestions, @jemunro !

Yes, I’m still using DSL1. I modified the channels following your suggestion and discovered nextflow was expecting the `interval` channel to contain paths to files, not to directories as I intended. So the final channels set is:
``` 
intervals = Channel.fromPath(cnv_out + "/*model/", type: 'dir', checkIfExists: true)
sample = Channel.fromPath(readcount_out + "/*.tsv", checkIfExists: true) `
```

The expected Channels output is:
```
[julianajose/project_cnv/data/intermediate/02.CollectReadCounts/sample1.tsv, julianajose/project_cnv/data/intermediate/04.CNVcall/cohort_01_of_56_model/]
```

However, the same error persists:
``` 
ERROR ~ No signature of method: groovyx.gpars.dataflow.DataflowQueue.getFileSystem() is applicable for argument types: () values: []
```

**Reply by jemunro** [reply]
Maybe try checking what the channel is actually emitting by using `view`:
```nextflow
ChannelCombined = sample.combine(intervals).view { "ChannelCombined: $it" }
```
